query/src/main/scala/filodb/query/exec/SetOperatorExec.scala

```diff
@@ -133,39 +155,72 @@ final case class SetOperatorExec(queryContext: QueryContext,
             rhsRows.close()
           }
         }
-        result += IteratorBackedRangeVector(lhs.key, rows)
+        val arrayBuffer = result.getOrElse(jk, ArrayBuffer())
+        val resRv = IteratorBackedRangeVector(lhs.key, rows)
+        if (index >= 0) arrayBuffer.update(index, resRv) else arrayBuffer.append(resRv)
+        result.put(jk, arrayBuffer)
       } else if (jk.isEmpty) {
         // "up AND ON (dummy) vector(1)" should be equivalent to up as there's no dummy label
-        result += lhs
+        val arrayBuffer = result.getOrElse(jk, ArrayBuffer())
+        arrayBuffer.append(lhs)
+        result.put(jk, arrayBuffer)
       }
     }
-    result.toList
+    result.values.flatten
   }
+  // scalastyle:on method.length
 
   private def setOpOr(lhsRvs: List[RangeVector]
```

Comment:
This is outside the scope of this PR, but good to add while we are at it. Do you mind adding English description of what each of AND, OR and UNLESS must do along with nuances? This helps because we keep forgetting and it takes some reverse engineering effort to recall - docs will make it easier. Thanks much!

Reply:
Added description